Rocket.Chat/Controllers/Chat/ChatViewController.swift

```diff
@@ -28,6 +28,22 @@ final class ChatViewController: SLKTextViewController {
     @IBOutlet weak var buttonScrollToBottom: UIButton!
     var buttonScrollToBottomMarginConstraint: NSLayoutConstraint?
 
+    var showButtonScrollToBottom: Bool = false {
+        didSet {
+            if showButtonScrollToBottom {
+                buttonScrollToBottomMarginConstraint?.constant = -64
+            } else {
+                buttonScrollToBottomMarginConstraint?.constant = 50
+            }
+
+            if showButtonScrollToBottom != oldValue {
+                UIView.animate(withDuration: 0.5) {
+                    self.view.layoutIfNeeded()
+                }
```

Comment:
This is working great, the position is now being calculated correctly, but this line is causing a very weird animation. Maybe if you change from `self.view` to `self.buttonScrollToBottom` this won't happen.

![2017-10-09 06 43 49](https://user-images.githubusercontent.com/551004/31332644-8d34f6f8-acbd-11e7-82b9-ac6ddeacd0ee.gif)